recipes/cc65/all/conanfile.py

```diff
@@ -0,0 +1,181 @@
+                mock_settings.arch = "x86"
+
+        original_settings = self.settings
+        try:
+            self.settings = mock_settings
+            yield
+        finally:
+            self.settings = original_settings
+
+    def _build_msvc(self):
+        msbuild = MSBuild(self)
+        msvc_platforms = {
+            "x86": "Win32",
+        }
+
+        msbuild.build(os.path.join(self._source_subfolder, "src", "cc65.sln"),
+                      build_type="Release", platforms=msvc_platforms)
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self._source_subfolder, "libsrc")):
+            autotools.make()
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    @property
+    def _make_args(self):
+        datadir = self._datadir
+        prefix = self.package_folder
+        samplesdir = self._samplesdir
+        if tools.os_info.is_windows:
+            datadir = tools.unix_path(datadir)
+            prefix = tools.unix_path(prefix)
+            samplesdir = tools.unix_path(samplesdir)
+        args = [
+            "PREFIX={}".format(prefix),
+            "datadir={}".format(datadir),
+            "samplesdir={}".format(samplesdir),
+        ]
+        if self.settings.os == "Windows":
+            args.append("EXE_SUFFIX=.exe")
+        return args
+
+    def _build_autotools(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            autotools.make(args=self._make_args)
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        if self.settings.compiler == "Visual Studio":
+            with tools.chdir(os.path.join(self._source_subfolder, "src")):
+                for fn in os.listdir("."):
+                    if not fn.endswith(".vcxproj"):
+                        continue
+                    tools.replace_in_file(fn, "v141", msvs_toolset(self.settings))
+                    tools.replace_in_file(fn, "<WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>", "")
+        if self.settings.os == "Windows":
+            # Add ".exe" suffix to calls from cl65 to other utilities
+            for fn, var in (("cc65", "CC65"), ("ca65", "CA65"), ("co65", "CO65"), ("ld65", "LD65"), ("grc65", "GRC")):
+                v = "{},".format(var).ljust(5)
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "cl65", "main.c"),
+                                      "CmdInit (&{v} CmdPath, \"{n}\");".format(v=v, n=fn),
+                                      "CmdInit (&{v} CmdPath, \"{n}.exe\");".format(v=v, n=fn))
+
+    def build(self):
+        self._patch_sources()
+        with self._mock_settings():
```

Comment:
Mocking is only needed for visual studio, right?

Reply:
Yes